spotifycli/test/test_spotifycli.py

```diff
@@ -0,0 +1,163 @@
+from io import StringIO
+from pathlib import Path
+from subprocess import PIPE
+from unittest import skip
+from unittest.mock import patch
+
+import dbus
+import dbusmock
+
+from spotifycli.spotifycli import main
+from spotifycli.version import __version__
+
+
+class TestSpotifyCLI(dbusmock.DBusTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.start_session_bus()
+        cls.dbus_con = cls.get_dbus(system_bus=False)
+
+    def setUp(self):
+        self.dbus_spotify_server_mock = self.spawn_server(
+            "org.mpris.MediaPlayer2.spotify",
+            "/org/mpris/MediaPlayer2",
+            "org.freedesktop.DBus.Properties",
+            system_bus=False,
+            stdout=PIPE
+        )
+        self.dbus_spotify_interface_mock = dbus.Interface(
+            self.dbus_con.get_object(
+                'org.mpris.MediaPlayer2.spotify', '/org/mpris/MediaPlayer2'
+            ),
+            dbusmock.MOCK_IFACE
+        )
+        metadata_file = Path(__file__).resolve().parent / 'dbus_metadata_response.py'
+        self.dbus_spotify_interface_mock.AddMethod(
+            '', 'Get', 'ss', 'v',
+            metadata_file.read_text(),
+        )
+        # TODO: Mock up the controls bus/interface, since I think it is slightly different
+        # self.dbus_spotify_interface_mock.AddMethod(
+        #     '', 'Play', '', '', ''
+        # )
```

Comment:
For each unit test run, we spin up the actual spotify dbus session server and build up an interface where we can stand up mock methods to be callable.  We then use `AddMethod` to add a single `Get` operation that we can use to read current metadata, etc.  In this case the return value from this dbus call will actually be the result of executing that little Python script above and grabbing the value from the `ret` variable.  Pretty nice, but could be better.

## What are the other args in AddMethod?

The second argument is the name of the method to add.  The third is the argument type being passed into the dbus call.  In this case, it is two strings, as found in this line: https://github.com/pwittchen/spotify-cli-linux/blob/43143c20dc40af61dcb70c303bced5edd83c042c/spotifycli/spotifycli.py#L293

These arguments are marshaled as strings (`'ss'`) and placed into the `args` array when the Python script above is called.  That is why we check `args[1]`, because that will contain the `specific_property` we are asking for.

The fourth argument is a `v`, because it is currently marshaled as a variant.  In this option, the Dbus package must dissect each variable type recursively and build out the dbus message with each argument type defined before its value.  If we can figure out how to do this part a little better, this is how we can fix the numeric/string/list changes I had to make to get it working.